pkg/watermark/fetch/edge_fetcher.go

```diff
@@ -71,15 +71,27 @@ func (e *edgeFetcher) GetWatermark(inputOffset isb.Offset) wmb.Watermark {
 	var epoch int64 = math.MaxInt64
 	var allProcessors = e.processorManager.GetAllProcessors()
 	for _, p := range allProcessors {
+		// headOffset is used to check whether this pod can be deleted.
+		headOffset := int64(-1)
+		// iterate over all the timelines of the processor and get the smallest watermark
 		debugString.WriteString(fmt.Sprintf("[Processor: %v] \n", p))
-		var t = p.GetOffsetTimeline().GetEventTime(inputOffset)
-		if t == -1 { // watermark cannot be computed, perhaps a new processing unit was added or offset fell off the timeline
-			epoch = t
-		} else if t < epoch {
-			epoch = t
+		for _, tl := range p.GetOffsetTimelines() {
+			var t = tl.GetEventTime(inputOffset)
```

Comment:
`inputOffset` is from a particular partition, while `GetOffsetTimelines()` returns OT for different partitions, is that correct? If yes, is this logic correct?